gui/melt-remainder.lua

```diff
@@ -0,0 +1,119 @@
+-- Add a display for how much melted metal is stored inside smelters.
+--@module = true
+--@enable = true
+--[====[
+
+gui/melt-remainder
+==================
+When enabled, a line of text is added to the Tasks screen of melters and magma smelters
+that shows how much metal is "stored" as a result of melting items in it. (The base game
+has no display of this information,)
+
+Click on the text for more detailed information.
+
+]====]
+
+local overlay = require("plugins.overlay")
+local widgets = require("gui.widgets")
+-- the existence of this is not documented anywhere of course
+local dialogs = require("gui.dialogs")
+
+ENABLED = ENABLED or false
+
+local function i_hate_lua(tbl)
+  local worst_language = 0
+  for _,_ in pairs(tbl) do
+    worst_language = worst_language + 1
+  end
+  return worst_language
+end
+
+local function get_melt_remainders(smelter)
+  if not smelter.melt_remainder then return nil end
+  local fractions = {}
+  local mat_count = #df.global.world.raws.inorganics
+  for i = 0, mat_count - 1 do
```

Comment:
although `#smelter.melt_remainder` *should* be the same as `#df.global.world.raws.inorganics`, it would be more consistent to define `mat_count = #smelter.melt_remainder` since you are indexing into `smelter.melt_remainder` in the loop.